src/boot/efi/efi-string.c

```diff
@@ -141,13 +140,16 @@ DEFINE_STRNDUP(char, xstrndup8, strnlen8);
 DEFINE_STRNDUP(char16_t, xstrndup16, strnlen16);
 
 int efi_memcmp(const void *p1, const void *p2, size_t n) {
+        const uint8_t *up1 = p1, *up2 = p2;
+        int r;
+
         if (!p1 || !p2)
                 return CMP(p1, p2);
 
-        const uint8_t *up1 = p1, *up2 = p2;
         while (n > 0) {
-                if (*up1 != *up2)
-                        return *up1 - *up2;
+                r = CMP(*up1, *up2);
```

Comment:
Same here? Also, I'd prefer to declare the `int r` here at least.

Reply:
Here, `*up1` and `*up2` are unsigned. So, I'd like to use `CMP()` for safety.
Also, we usually declare `r` at the beginning, as it is a kind of special variable.